railties/test/generators/action_text_generator_test.rb

```diff
@@ -0,0 +1,79 @@
+# frozen_string_literal: true
+
+require "minitest/mock"
+require "isolation/abstract_unit"
+require "generators/generators_test_helper"
+require "generators/action_text/install/install_generator"
+
+class ActionTextGeneratorTest < Rails::Generators::TestCase
+  include ActiveSupport::Testing::Isolation
+  tests ActionText::Generators::InstallGenerator
+
+  def setup
+    build_app
+    Rails.application = TestApp::Application
+  end
+
+  def teardown
+    teardown_app
+  end
+
+  def test_generators_install_javascript_dependencies
+    Dir.chdir(app_template_path) do
+      js_dependencies_stub = { "trix"=>"latest", "@rails/actiontext"=>"latest" }
+      generator.stub(:js_dependencies, js_dependencies_stub) do
+        generator.stub(:rails_command, nil, ["app:binstub:yarn"]) do
+          quietly { generator.install_javascript_dependencies }
```

Comment:
@jonathanhefner When I use `run_generator` here I run into following error: 

```
Traceback (most recent call last):
	3: from /Users/abhaynikam/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/minitest-5.14.0/lib/minitest/parallel.rb:33:in `block (2 levels) in start'
	2: from /Users/abhaynikam/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/minitest-5.14.0/lib/minitest.rb:1026:in `run_one_method'
	1: from /Users/abhaynikam/personal/rails/activesupport/lib/active_support/testing/isolation.rb:23:in `run'
/Users/abhaynikam/personal/rails/activesupport/lib/active_support/testing/isolation.rb:23:in `load': marshal data too short (ArgumentError)
Traceback (most recent call last):
	3: from /Users/abhaynikam/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/minitest-5.14.0/lib/minitest/parallel.rb:33:in `block (2 levels) in start'
	2: from /Users/abhaynikam/.rbenv/versions/2.7.0/lib/ruby/gems/2.7.0/gems/minitest-5.14.0/lib/minitest.rb:1026:in `run_one_method'
	1: from /Users/abhaynikam/personal/rails/activesupport/lib/active_support/testing/isolation.rb:23:in `run'
/Users/abhaynikam/personal/rails/activesupport/lib/active_support/testing/isolation.rb:23:in `load': marshal data too short (ArgumentError)
```

I have seen at a lot of places(app_generator_test) in the codebase using either 
1. `generator.invoke_all`
2. `generator.method_name`
3. `run_generator`

I preferred to use `generator.method_name` to avoid running `yarn add` multiple times(stubbed) and slowing the test suite. Will it be okay if we go with `generator.method_name`? If not, I'll try to invest more time in it. 😊

Reply:
@jonathanhefner Using `run_generator` throws:
```
lib/active_support/testing/isolation.rb:23:in `load': marshal data too short (ArgumentError)
```

Any thoughts on how we could resolve it? I spent too much time on it but not able to fix it and use `run_generator` instead of `generator.method_name`.